website/docs/d/workspace.html.markdown

```diff
@@ -0,0 +1,53 @@
+---
+layout: "tfe"
+page_title: "Terraform Enterprise: tfe_workspace"
+sidebar_current: "docs-datasource-tfe-workspace"
+description: |-
+  Get information on a workspace.
+---
+
+# Data Source: tfe_workspace
+
+Use this data source to get information about a workspace.
+
+## Example Usage
+
+```hcl
+data "tfe_workspace" "test" {
+  name = "my-workspace-name"
+  organization = "my-org-name"
```

Comment:
Hmm, I'm a little concerned about this interface being unwieldy, but I'm not sure whether the thing I want is possible or not...

My use case for this is shown in [this example policies config](https://github.com/hashicorp/tfe-policies-example/blob/master/main.tf). Basically I have a bunch of policy set resources, and each one needs to provide a bunch of workspace IDs, which I'd rather access by name. But with per-workspace data sources, I'd need to write four extra lines of configuration for every name I want to use, and I'd still have the problem of double-bookkeeping, where I have to keep my set of `tfe_workspace` data sources up to date. That's not too onerous in that toy example, but in a real-life situation with 50+ workspaces instead of just 5, it's kind of brutal. 

Would it be possible to also make a `tfe_workspaces` or `tfe_workspace_ids` data source, which would have an `ids` attribute with a _map_ value? Or is that not feasible because of something about the way providers work?

Reply:
I think a data source can have an `ids` attribute which contains a `map` of workspace names and their external IDs. But in that case I think the question is what kind of filter you want to use to select which IDs should be added to the map?

If that filter requires you to specify all names of the workspaces you want to select, you end up with having to declare the the same list twice. Once in the `tfe_workspace_ids` data source and once in the `tfe_policy_set` resource. Yet I agree that this would still be better then the current solution for your use case.

But after looking at your example, I think there is a much better way to make this resource user friendly. I don't think we should mix and match workspace IDs in TFE resources. So instead of using the external IDs, we should probably just use the workspace names here.

Exporting the external ID of a workspace is only done for specific use cases (see #19 for more details), so that ID should itself not be used in other resources. I guess this one slipped through when reviewing the PR.

So I'll work on a PR to update the `tfe_policy_set` resource today so you can have a look at it when you get online. I think that would be a much better solution that using a data source with a map.

Reply:
Please checkout PR #44 and then mainly the updated docs as those will show the UX changes when using the resource: https://github.com/terraform-providers/terraform-provider-tfe/pull/44/files#diff-237abdf1e9ce57c21537937ad3e09eea